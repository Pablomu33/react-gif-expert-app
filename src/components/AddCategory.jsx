import { useState } from "react"
import PropTypes from "prop-types"

export const AddCategory = ({onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChange =({target})=>{

        setInputValue(target.value)
    }

    const onSubmit= (event)=>{
        event.preventDefault()
       // console.log(inputValue)
      if(inputValue.trim().length<= 1 ) return;

      onNewCategory(inputValue.trim()) 
        //setCategories (categories => [ inputValue, ...categories])
        setInputValue('')
    }
  return (
    
    <form onSubmit={ onSubmit} aria-label="form" >
      <input
      type="text"
      placeholder="buscar Gif"
      value= {inputValue}
      onChange={ onInputChange }
      />
      
    </form>
            
      )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}