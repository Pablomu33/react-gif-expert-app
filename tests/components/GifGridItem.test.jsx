import { render } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"
import { screen, configure } from '@testing-library/react'

describe('pruebas en <GifItem/>', ()=>{

    const title = 'saitama'
    const url = 'https://one-punch.com/saitamam.jpg'

test('debe hace match con el snapshot', () => { 

    const {container} = render( < GifGridItem title={title} url= {url} />)
    expect(container).toMatchSnapshot()
})

test('should mostrar la imagen con el url y el alt', () => { 

    render(<GifGridItem title={title} url= {url} />);
    const{src, alt} = screen.getByRole('img');
    expect(src).toBe(url)
    expect(alt).toBe(alt)


    //expect(screen.getByRole('img').src).toBe(url);
 })

 test('mostrar el titulo del componente', () => { 

    render(<GifGridItem title={title} url= {url} />);
    expect(screen.getByAltText(title).toBeTruthly)
  })

})
