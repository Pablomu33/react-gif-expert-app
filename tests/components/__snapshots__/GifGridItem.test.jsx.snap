// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem/> debe hace match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://one-punch.com/saitamam.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
